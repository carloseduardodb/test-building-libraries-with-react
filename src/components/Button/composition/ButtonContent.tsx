import React from 'react';

interface ButtonContentProps {
  text: string;
  type: 'xs' | 'md' | 'lg';
}

export const ButtonContent = ({ text, type }: ButtonContentProps): any => {
  let buttonClass = '';

  switch (type) {
    case 'xs':
      buttonClass = 'text-xs';
      break;
    case 'md':
      buttonClass = 'text-md';
      break;
    case 'lg':
      buttonClass = 'text-lg';
      break;
    default:
      break;
  }
  return <div className={buttonClass}>{text}</div>;
};
