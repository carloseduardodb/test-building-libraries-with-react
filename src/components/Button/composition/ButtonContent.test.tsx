import renderer from 'react-test-renderer';
import React from 'react';
import { ButtonContent } from './ButtonContent';

describe('ButtonContent', () => {
  it('test_render_with_text_and_type_props', () => {
    const tree = renderer
      .create(<ButtonContent text="Hello World" type="md" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('test_render_with_xs_type_prop', () => {
    const tree = renderer
      .create(<ButtonContent text="Hello World" type="xs" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('test_render_with_md_type_prop', () => {
    const tree = renderer
      .create(<ButtonContent text="Hello World" type="md" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('test_render_with_empty_text_prop', () => {
    const tree = renderer.create(<ButtonContent text="" type="md" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('test_render_with_invalid_type_prop', () => {
    const value: any = 'invalid';
    const tree = renderer
      .create(<ButtonContent text="Hello World" type={value} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
