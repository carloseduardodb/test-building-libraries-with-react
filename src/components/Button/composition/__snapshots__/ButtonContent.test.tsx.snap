// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContent test_render_with_empty_text_prop 1`] = `
<div
  className="text-md"
/>
`;

exports[`ButtonContent test_render_with_invalid_type_prop 1`] = `
<div
  className=""
>
  Hello World
</div>
`;

exports[`ButtonContent test_render_with_md_type_prop 1`] = `
<div
  className="text-md"
>
  Hello World
</div>
`;

exports[`ButtonContent test_render_with_text_and_type_props 1`] = `
<div
  className="text-md"
>
  Hello World
</div>
`;

exports[`ButtonContent test_render_with_xs_type_prop 1`] = `
<div
  className="text-xs"
>
  Hello World
</div>
`;

exports[`renders correctly 1`] = `
<button
  onClick={[Function]}
  type="button"
>
  Clicked this many times 0
</button>
`;
